nixos/tests/networking.nix

```diff
@@ -380,6 +380,50 @@ let
               router.wait_until_succeeds("ping -c 1 192.168.1.3")
         '';
     };
+    fou = {
+      name = "foo-over-udp";
+      nodes.machine = { ... }: {
+        virtualisation.vlans = [ 1 ];
+        networking = {
+          useNetworkd = networkd;
+          useDHCP = false;
+          interfaces.eth1.ipv4.addresses = mkOverride 0
+            [ { address = "192.168.1.1"; prefixLength = 24; } ];
+          fous = {
+            fou1 = { port = 9001; };
+            fou2 = { port = 9002; protocol = 41; };
+            fou3 = mkIf (!networkd)
+              { port = 9003; local.address = "192.168.1.1"; };
+            fou4 = mkIf (!networkd)
+              { port = 9004; local = { address = "192.168.1.1"; dev = "eth1"; }; };
+            fou5 = mkIf (!networkd)
+              { port = 9005; peer = { address = "192.168.1.2"; port = 9105; }; };
+          };
+        };
+        systemd.services =
+          let
+            after = optional (!networkd) "network-addresses-eth1.service";
+          in
+            {
+              fou3-fou-encap.after = after;
+              fou5-fou-encap.after = after;
+            };
+      };
+      testScript = { ... }:
+        ''
+          machine.wait_for_unit("network.target")
+          machine.succeed("ip addr >&2")
+          fous = machine.succeed("ip fou show")
```

Comment:
You are aware of the JSON output option of `ip`? You could make this check "proper". This is likely good enough but just in case you want to go the extra mile.

Reply:
nope, hadn't ever noticed that! looks like a much better way to do this.